apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -655,6 +663,19 @@ init_resource(Module, OnCreate, ResId, Config) ->
                                  status = #{is_alive => true}},
     emqx_rule_registry:add_resource_params(ResParams).
 
+init_resource_with_retrier(Module, OnCreate, ResId, Config) ->
+    try
+        Params = Module:OnCreate(ResId, Config),
+        ResParams = #resource_params{id = ResId,
+                                    params = Params,
+                                    status = #{is_alive => true}},
+        emqx_rule_registry:add_resource_params(ResParams)
+    catch Class:Reason:ST ->
+        Interval = persistent_term:get({emqx_rule_engine, resource_restart_interval}, ?T_RETRY),
+        emqx_rule_monitor:ensure_resource_retrier(ResId, Interval),
```

Comment:
if we read the interval when needed (instead of storing it in the looping state), then it can be reconfigured on the fly.

Reply:
ok, fair point.